flax/linen/recurrent.py

```diff
@@ -250,6 +250,7 @@ def _concat_dense(inputs: Array,
       inputs, kernel, bias = promote_dtype(inputs, kernel, bias, dtype=self.dtype)
       y = jnp.dot(inputs, kernel)
       if use_bias:
+        assert bias is not None
```

Comment:
I don't love having to add seemingly redundant asserts for mypy... but we can live with a few.

Reply:
maybe add a comment for these asserts that it's to satisfy typecheckers?

Reply:
Will do.